java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -1252,15 +1322,18 @@ public static void writeConcatedStream(SerializedTableHeader[] headers,
                                          OutputStream out) throws IOException {
     ColumnBufferProvider[][] providers = providersFrom(headers, dataBuffers);
     SerializedTableHeader combined = calcConcatedHeader(providers);
```

Comment:
Given the writer was moved within the try block, curious why this was not as well.  It would cover the unlikely case an error is thrown during  `calcConcatedHeader`.